run-python/runner.go

```diff
@@ -0,0 +1,187 @@
+package runner
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/stoic-cli/stoic-cli-core"
+	shell "github.com/stoic-cli/stoic-cli-core/run-shell"
+	"github.com/stoic-cli/stoic-cli-core/tool"
+)
+
+func newRunner(s stoic.Stoic, t stoic.Tool) (tool.Runner, error) {
+	var options PythonOptions
+	err := mapstructure.Decode(t.Config().Runner.Options, &options)
+	if err != nil {
+		return nil, err
+	}
+
+	python, ok := t.Config().Runner.Options["python"].(string)
```

Comment:
Could we defer this kind of logic to the config?

Reply:
What do you mean?

This function is the first point the Runner and the Config meet. No validation is performed by the engine.

One thing I was considering, was to somehow expose the `*Options` types from `tool.Runner`s and `tool.Getter`s, so the engine can handle both loading and errors. I haven't played too much on that front, but I think this is something that will see some changes, going forward.

Reply:
Sorry for the bad explanation, but I was thinking the same as you mentioned, handling the extraction of config material in a separate file, including casting etc